Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,183 @@
+                        texture.Apply();
+
+                        var savePath = GetSavePath(mat, textureName);
+                        TextureData.WriteToFile(savePath, texture.EncodeToPNG());
+
+                        // Setting the texture can fail if the asset database has not imported it yet.
+                        // If that happens, save it for later and try again:
+                        if (!SetSerializedTextureForMaterial(mat, textureName)) {
+                            pendingMaterialUpdates.Add(new Tuple<Material, string>(mat, textureName));
+
+                            // Set the texture on the material to null for now so the render texture
+                            // is not saved to file again if this gets called again before it was
+                            // set to the texture in the asset library:
+                            mat.SetTexture(textureName, null);
+                        }
+                    }
+                }
+            }
+        }
+
+        private bool SetSerializedTextureForMaterial(Material mat, string textureName) {
+            var savePath = GetSavePath(mat, textureName);
+
+            // LoadAssetAtPath can throw an exception if it's called during a domain backup but we have no way to detect that:
+            try {
+                if (AssetDatabase.LoadAssetAtPath<Texture2D>(savePath) == null) {
+                    EditorApplication.delayCall -= UpdatePendingMaterials;
+                    EditorApplication.delayCall += UpdatePendingMaterials;
+
+                    return false;
+                }
+            }
+            catch (Exception ex) {
+                return false;
+            }
+
+            // Ensure the texture is up to date:
+            AssetDatabase.ImportAsset(savePath);
+
+            var savedTexture = AssetDatabase.LoadAssetAtPath<Texture2D>(savePath);
+            
+            mat.SetTexture(textureName, savedTexture);
+
+            if (textureName == _MetallicGlossMap ||
+                textureName == _MaskMap) {
+                mat.EnableKeyword(_METALLICGLOSSMAP);
+                mat.EnableKeyword(_METALLICSPECGLOSSMAP);
+
+                TextureImporter importer = (TextureImporter)AssetImporter.GetAtPath(savePath);
+                if (importer != null) {
+                    importer.sRGBTexture = false;
+                    importer.alphaIsTransparency = false;
+                    importer.ignorePngGamma = true;
+                }
+            }
+
+            return true;
+        }
+
+        private void UpdatePendingMaterials() {
+            EditorApplication.delayCall -= UpdatePendingMaterials;
+
+            for (int i = pendingMaterialUpdates.Count - 1; i >= 0; i--) {
+                Tuple<Material, string> pendingMaterial = pendingMaterialUpdates[i];
+                if (SetSerializedTextureForMaterial(pendingMaterial.Item1, pendingMaterial.Item2)) {
```

Comment:
It's a bit complicated. I've added some more comments.
This code runs when the meshsync server is saved. It looks at all textures on its materials that might be baked to a render texture. 
If it finds one that is a render texture, it saves that to the asset database. The asset database works asynchronously and could import the texture at some later time so the call to `LoadAssetAtPath` may return null at the time the server is saved.

To get around that, if the texture does not yet exist in the asset database, some info is added to the `pendingMaterialUpdates` list and a call to `UpdatePendingMaterials` scheduled for the next update.
When `UpdatePendingMaterials` gets called again it checks if `LoadAssetAtPath` returns a texture and if it does, it assigns the texture to the material. If it does not exist, a new call to UpdatePendingMaterials is scheduled.

`UpdatePendingMaterials` always unsubscribes itself at the start to avoid an infinite loop. The calls to `SetSerializedTextureForMaterial` may schedule the delayed call again. There is a safety unsubscribe there to ensure it is only scheduled once.

Reply:
Might be better to have the caller do the registration, as suggested in another comment

Reply:
`SetSerializedTextureForMaterial` is called from multiple places, I'm trying to avoid code duplication.

Reply:
If you add new function to the `EditorApplication.delayCall` callbacks, during the process of the `EditorApplication.delayCall` callbacks running, does that new function get called in the current loop?

Reply:
i.e I'm picturing a scenario where there's a delaycall callback to `UpdatePendingMaterials`. This adds another `UpdatePendingMaterials` callback to the delaycall callbacks list, so when the first function finishes the second one runs immediately.

Reply:
No, they get called in the next delayed call.